src/sentry/auth/services/auth/impl.py

```diff
@@ -49,9 +50,19 @@ def enable_partner_sso(
         sender: str | None = None,
     ) -> None:
         with enforce_constraints(transaction.atomic(router.db_for_write(AuthProvider))):
-            auth_provider_query = AuthProvider.objects.filter(
-                organization_id=organization_id, provider=provider_key, config=provider_config
-            )
+            if provider_key == "fly":
```

Comment:
Is there a way to not hardcode this?
Otherwise LGTM.

Reply:
rn fly is the only one we'd do this for, but we could probably set up some sort of mapping of equivalent providers